File: linkedin-automator/backend/routers/posts.py
from fastapi import APIRouter, HTTPException, Query

from models.schemas import GeneratePostsRequest, UpdatePostContentRequest
from services import rag
from services.post_generator import generate_posts

router = APIRouter(prefix="/posts", tags=["posts"])


@router.post("/generate")
async def generate(body: GeneratePostsRequest):
    posts = await generate_posts(body.transcript_id, body.transcript_text)
    return {"posts": posts}


@router.get("")
def list_posts(
    transcript_id: str | None = Query(None),
    status: str | None = Query(None),
):
    return rag.get_posts(transcript_id=transcript_id, status=status)


@router.get("/{post_id}")
def get_post(post_id: str):
    try:
        return rag.get_post(post_id)
    except Exception:
        raise HTTPException(status_code=404, detail="Post not found")


@router.patch("/{post_id}")
def update_post(post_id: str, body: UpdatePostContentRequest):
    if body.lang not in ("es", "en", "zh"):
        raise HTTPException(status_code=400, detail="lang must be es, en, or zh")
    rag.update_post_content(post_id, body.lang, body.content)
    return {"ok": True}

File: linkedin-automator/frontend/src/api/posts.ts
import axios from 'axios';
import type { Post, Lang } from '../types';

const api = axios.create({ baseURL: 'http://localhost:8000/api' });

export async function generatePosts(
  transcriptId: string,
  transcriptText: string,
): Promise<Post[]> {
  const { data } = await api.post<{ posts: Post[] }>('/posts/generate', {
    transcript_id: transcriptId,
    transcript_text: transcriptText,
  });
  return data.posts;
}

export async function listPosts(transcriptId?: string): Promise<Post[]> {
  const params: Record<string, string> = {};
  if (transcriptId) params.transcript_id = transcriptId;
  const { data } = await api.get<Post[]>('/posts', { params });
  return data;
}

export async function getPost(id: string): Promise<Post> {
  const { data } = await api.get<Post>(`/posts/${id}`);
  return data;
}

export async function updatePostContent(
  id: string,
  lang: Lang,
  content: string,
): Promise<void> {
  await api.patch(`/posts/${id}`, { lang, content });
}
